crates/matrix-sdk/src/sliding_sync.rs

```diff
@@ -339,23 +334,102 @@ impl SlidingSyncRoom {
             self.prev_batch.lock_mut().replace(batch.clone());
         }
 
-        if !timeline.is_empty() {
+        // There is timeline updates.
+        if !timeline_updates.is_empty() {
+            // If we come from a cold storage, we overwrite the timeline queue with the
+            // timeline updates.
             if self.is_cold.load(Ordering::SeqCst) {
-                // if we come from cold storage, we hard overwrite
-                self.timeline.lock_mut().replace_cloned(timeline);
+                self.timeline_queue.lock_mut().replace_cloned(timeline_updates);
                 self.is_cold.store(false, Ordering::SeqCst);
-            } else if *limited {
-                // the server alerted us that we missed items in between
-                self.timeline.lock_mut().replace_cloned(timeline);
-            } else {
-                let mut ref_timeline = self.timeline.lock_mut();
-                for e in timeline {
-                    ref_timeline.push_cloned(e);
+            }
+            // The server alerted us that we missed items in between.
+            else if *limited {
+                self.timeline_queue.lock_mut().replace_cloned(timeline_updates);
+            }
+            // It's the hot path. We have new updates that must be added to the existing timeline
+            // queue.
+            else {
```

Comment:
I find this formatting a bit unusual, maybe move the comments inside the `if` / `else` bodies? I.e. like it was before ^^

Reply:
Putting the comment on the `if` or `else` blocks means it's documenting the block. Otherwise if we put the comment inside, we can think it documents the first instruction of the block. That's my interpretation. But I'm fine reverting it. Doing it now.